lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -1199,6 +1341,89 @@ pub enum TdpErrCode {
     AE = 3,
 }
 
+#[derive(Debug, Clone)]
+pub struct SharedDirectoryWriteRequest {
+    completion_id: u32,
+    directory_id: u32,
+    offset: u64,
+    path_length: u32,
+    path: String,
+    // write_data: Vec<u8>,
```

Comment:
Shouldn't we use an owning Vec in the Rust version and only use ptr+len in the C version?

Reply:
Yeah! I've changed it to `Vec<u8>`